pep-0675.rst

```diff
@@ -913,6 +936,275 @@ is documentation, which is easily ignored and often not seen. With
 ``Literal[str]``, API misuse requires conscious thought and artifacts
 in the code that reviewers and future developers can notice.
 
+.. _appendix_C:
+
+Appendix C: ``str`` methods that preserve ``Literal[str]``
+==========================================================
+
+The ``str`` class has several methods that would benefit from
+``Literal[str]``. For example, users might expect
+``"hello".capitalize()`` to have the type ``Literal[str]`` similar to
+the other examples we have seen in the `Inferring Literal[str]
+<inferring_literal_str>`_ section. Inferring the type ``Literal[str]``
+is correct because the string is not an arbitrary user-supplied string
+- we know that it has the type ``Literal["HELLO"]``, which is
+compatible with ``Literal[str]``. In other words, the ``capitalize``
+method preserves the ``Literal[str]`` type. There are several other
+``str`` methods that preserve ``Literal[str]``.
+
+We propose updating the stub for ``str`` in typeshed so that the
+methods are overloaded with the ``Literal[str]``-preserving
+versions. This means type checkers do not have to hardcode
+``Literal[str]`` behavior for each method. It also lets us easily
+support new methods in the future by updating the typeshed stub.
+
+For example, to preserve literal types for the ``capitalize`` method,
+we would change the stub as below:
+
+::
+
+    # before
+    def capitalize(self) -> str: ...
+
+    # after
+    @overload
+    def capitalize(self: Literal[str]) -> Literal[str]: ...
+    @overload
+    def capitalize(self) -> str: ...
```

Comment:
Don't know if that's the right place for it to ask.

Instead of overloads, wouldn't it also work / be better to use a bound TypeVar or the new `Self` type? The way I understand the PEP, `LiteralString` should be defined as subtype of `str` within typeshed.

From the PEP:
```txt
Rationale

[...] LiteralString is the "supertype" of all literal string types.
In effect, this PEP just introduces a type in the type hierarchy
between Literal["foo"] and str. [...]
```

Reply:
A `TypeVar` wouldn't work for this case, sadly, as the relationship here doesn't generalise to other subtypes of `str`:

```python
>>> class Foo(str): ...
... 
>>> Foo('hi')
'hi'
>>> type(_)
<class '__main__.Foo'>
>>> Foo('hi').capitalize()
'Hi'
>>> type(_)
<class 'str'>

```

Reply:
I think a TypeVar with constraints (similar to AnyStr) might work?

Reply:
Hmm — that might work, but might also run into mypy bugs similar to https://github.com/python/typeshed/pull/6762?

Reply:
> Don't know if that's the right place for it to ask.

@cdce8p You could also bring it up on the official typing-sig thread for this PEP, which appears to be [this one](https://mail.python.org/archives/list/typing-sig@python.org/thread/VB74EHNM4RODDFM64NEEEBJQVAUAWIAW/)—that might be more generally visible than a comment on a merged PR, if you're looking for a wider discussion.

On a more editorial note, it looks like the PEP hasn't yet been updated with the link to the actual discussion thread, which I only found after a fair amount of determined digging (and guessing). I'll open a PR to add that.